pkg/k8sclient/k8sclient.go

```diff
@@ -53,15 +55,15 @@ func SetPodAnnotation(pod v1.Pod, key string, value string) error {
 	if err != nil {
 		return err
 	}
-	patchData := make([]patch, 1)
-	patchData[0].Op = "add"
-	patchData[0].Path = "/metadata/annotations/" + key
-	newAnnotation := `"` + value + `"`
-	patchData[0].Value = json.RawMessage(newAnnotation)
-	jsonData, err := json.Marshal(patchData)
+
+	merge := Update{}
+	merge.Meta.Annotations = make(map[string]json.RawMessage)
+	merge.Meta.Annotations[key] = json.RawMessage(`"` + value + `"`)
+
+	jsonData, err := json.Marshal(merge)
 	if err != nil {
 		return err
 	}
-	_, err = clientset.CoreV1().Pods(pod.ObjectMeta.Namespace).Patch(context.TODO(), pod.ObjectMeta.Name, types.JSONPatchType, jsonData, metav1.PatchOptions{})
+	_, err = clientset.CoreV1().Pods(pod.ObjectMeta.Namespace).Patch(context.TODO(), pod.ObjectMeta.Name, types.MergePatchType, jsonData, metav1.PatchOptions{})
```

Comment:
The switch from the `Patch` to the `Merge` type allows the annotation to be set when the pod currently has no annotations, otherwise it should behave identically